Source/Blazorise/Utilities/IO/RemoteFileEntryStreamReader.cs

```diff
@@ -81,5 +83,18 @@ public async Task WriteToStreamAsync( Stream stream, CancellationToken cancellat
                 }
             }
         }
+        
+        /// <summary>
+        /// If we're running on WebAssembly, then we should give time for the single thread to catch up and refresh the UI.
+        /// </summary>
+        /// <returns></returns>
+        private Task RefreshUI()
+        {
+            if ( OperatingSystem.IsBrowser() )
+                return Task.Delay( 1 );
```

Comment:
Have you tried with `Task.Yield` instead?

Reply:
I have and still works, however seems like it's 'tighter' and the ui will have more of a lagging feel.